### PR TITLE
Fixed enabling/disabling preferences

### LINKED ISSUE #3974: User interface options are not hidden when disable them in Admin settings
#### Software and hardware versions 
Collect v1.27.3, Android v5-10 probably all, device used...

#### Problem description
When testing #3933 I have noticed a regression issue that is visible on the master and store version. User interface options are not hiding after choosing them to hide in Admin Settings/User Settings.

#### Steps to reproduce the problem
1. Go to Admin Settings/User Settings
2. Uncheck options: Theme, language, text font
3. Go to General Settings/User interface
gif attached:
ezgif com-video-to-gif (13)

#### Expected behavior
After unselecting options in Admin Settings they should not be visible in General Settings.

#### Other information 
From my investigation, it looks like we lost it somewhere between 1.26 and 1.27.

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #3974

#### What has been done to verify that this works as intended?
I tested implemented changes manually and added automated tests.

#### Why is this the best possible solution? Were any other approaches considered?
It was just a bug we introduced refactoring preferences so I just fixed it and added tests.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
It just requires testing how settings work when they are enabled/disabled, nothing else should be affected.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/preferences/ExperimentalPreferencesFragment.java [hunk 1]
```diff
@@ -6,13 +6,12 @@
...
 import androidx.annotation.Nullable;
 import androidx.fragment.app.FragmentActivity;
-import androidx.preference.PreferenceFragmentCompat;
 
 import org.odk.collect.android.R;
 import org.odk.collect.android.activities.CollectAbstractActivity;
 import org.odk.collect.android.injection.DaggerUtils;
 
-public class ExperimentalPreferencesFragment extends PreferenceFragmentCompat {
+public class ExperimentalPreferencesFragment extends BasePreferenceFragment {
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/preferences/ExperimentalPreferencesFragment.java:14]: Ah nice. These were left over from when they were `androidx.` and everything else was just `android.`

File: collect_app/src/main/java/org/odk/collect/android/preferences/GeneralPreferencesFragment.java [hunk 2]
```diff
@@ -82,22 +82,21 @@ public void onCreate(final Bundle savedInstanceState) {
     public boolean onPreferenceClick(Preference preference) {
         if (MultiClickGuard.allowClick(getClass().getName())) {
             PreferenceFragmentCompat basePreferenceFragment = null;
-            boolean adminMode = getArguments().getBoolean(INTENT_KEY_ADMIN_MODE, false);
             switch (preference.getKey()) {
                 case "protocol":
-                    basePreferenceFragment = ServerPreferencesFragment.newInstance(adminMode);
+                    basePreferenceFragment = new ServerPreferencesFragment();
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/preferences/GeneralPreferencesFragment.java]: Are we not losing the `adminMode` argument here?

File: collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesFragment.java [hunk 3]
```diff
@@ -45,17 +47,15 @@
...
     public static final String ADMIN_PREFERENCES = "admin_prefs";
 
     @Override
     public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
-        super.onCreatePreferences(savedInstanceState, rootKey);
-
         getPreferenceManager().setSharedPreferencesName(ADMIN_PREFERENCES);
-
         setPreferencesFromResource(R.xml.admin_preferences, rootKey);
+        ((CollectAbstractActivity) getActivity()).initToolbar(getPreferenceScreen().getTitle());
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesFragment.java]: Can we do a conditional cast here like in `BasePreferencesFragment`?

File: collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesFragment.java [hunk 4]
```diff
@@ -144,36 +144,31 @@ public boolean onPreferenceClick(Preference preference) {
...
+    public static class MainMenuAccessPreferences extends PreferenceFragmentCompat {
         @Override
-        public void onCreate(Bundle savedInstanceState) {
-            super.onCreate(savedInstanceState);
+        public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
             getPreferenceManager().setSharedPreferencesName(ADMIN_PREFERENCES);
-
-            addPreferencesFromResource(R.xml.main_menu_access_preferences);
+            setPreferencesFromResource(R.xml.main_menu_access_preferences, rootKey);
+            ((CollectAbstractActivity) getActivity()).initToolbar(getPreferenceScreen().getTitle());
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesFragment.java]: Conditional cast for the child Fragments as well.

**Review by seadowg — CHANGES_REQUESTED** [review]
Looks great! Couple of small changes inline. 

I also think we could probably get rid of a few regression Espresso tests now we have this coverage. `uncheckedSettings_ShouldNotBeVisibleInGeneralSettings` in `UserSettingsTest` is probably not needed for example. Would you be able to have a look and see if there are any others we could get rid of? As we've been doing we might want to move the issue tag comment over to the fragment test.

**Comment by seadowg** [comment]
Oh and just a note that this will have conflicts with #3997 most likely as both PRs do some tidying up (moving to `setPreferencesFromResource`) but I didn't spot anything that's moving in a different direction so it should be ok to fix.

File: collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesFragment.java (hunk 3 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesFragment.java]: Now it doesn't exist there and we set title only in BasePreferenceFragment which works for every screen.

File: collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesFragment.java (hunk 4 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesFragment.java]: Now it doesn't exist there and we set title only in BasePreferenceFragment which works for every screen.

File: collect_app/src/main/java/org/odk/collect/android/preferences/GeneralPreferencesFragment.java (hunk 2 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/preferences/GeneralPreferencesFragment.java]: No I just pass arguments from activity to fragments https://github.com/getodk/collect/pull/4014/files#diff-fc03fbca20bb6f86edfdee031f3509a4R74

**Comment by grzesiek2010** [comment]
> Would you be able to have a look and see if there are any others we could get rid of? As we've been doing we might want to move the issue tag comment over to the fragment test.

I did that. I was able to identify 3 duplicated tests. 2 of them come from regression package:
- uncheckedSettings_ShouldNotBeVisibleInGeneralSettings
- showGuidance_shouldBeHidden

so a question to @mmarciniak90 @kkrawczyk123 how do you want to handle such a case where we can replace slower espresso tests which are part of your regression check list? Do you want to add a link/comment to my new tests which are replacements or what?

**Comment by grzesiek2010** [comment]
@seadowg I fixed conflicts so please add `needs testing` label if everything is fine.

**Review by seadowg — APPROVED** [review]

**Comment by kkrawczyk123** [comment]
@grzesiek2010 The issue has been fixed but very unfortunate regression has been introduced. Some of the options are disappearing after deselecting them in Admin Settings/User Settings:
- blank form update mode 
- periodic form updates check
- automatic download 
- hide old form version
- high res video 
- image size
- show guidance for questions 
- collect anonymous usage data
ezgif com-video-to-gif (17)
The issue does not occur on master, screenshot form master:
Screenshot (Aug 19, 2020 5_29_43 PM)

@getodk-bot unlabel "needs testing"

**Comment by grzesiek2010** [comment]
Ohh I fixed it and added tests.

**Comment by kkrawczyk123** [comment]
**Tested with success!**
Verified o Androids 5.1, 7.1 and 10.

Tested scenarios:
- enable and disable all of User Settings one by one
- enable and disable different combinations of User Sttings
- light mode/dark mode
- screen rotation
- minimize app
- regression check on Main Menu Settings and Form Entry Settings
- confirmed introduced regression has been fixed

**Comment by mmarciniak90** [comment]
Tested with success on Android versions 7.0, 8.1, 9.0

@getodk-bot unlabel "needs testing"
@getodk-bot label "behavior verified"

**Pull request merged**